### DISCUSSION TITLE
Gitlens fails after recent update (unknown field name: HEAD)

### DISCUSSION THREAD

**Discussion by mdear** [gitkraken/vscode-gitlens#1376] [Q&A]
After recently upgrading GitLens (which was working for me), I am now getting git failures.

I am using ssh-remote to CentOS 7 : Description:	CentOS Linux release 7.7.1908 (Core)
Linux cpc-inception 3.10.0-1062.el7.x86_64 #1 SMP Wed Aug 7 18:08:02 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
git version 1.8.3.1  (which is the version supported by my version of the OS, which I am not able to upgrade further).

Is there a way around this ?

Command failed: /usr/bin/git -c core.quotepath=false -c color.ui=false for-each-ref --format=%3ch%3e%(HEAD)%3cn%3e%(refname)%3cu%3e%(upstream:short)%3ct%3e%(upstream:track)%3cr%3e%(objectname)%3cd%3e%(committerdate:iso8601) refs/heads refs/remotes fatal: unknown field name: HEAD

Command failed: /usr/bin/git -c core.quotepath=false -c color.ui=false tag -l --format=%3cn%3e%(refname)%3c*r%3e%(*objectname)%3cr%3e%(objectname)%3cd%3e%(creatordate:iso8601)%3cad%3e%(authordate:iso8601)%3cs%3e%(subject) error: unknown option `format=%3cn%3e%(refname)%3c*r%3e%(*objectname)%3cr%3e%(objectname)%3cd%3e%(creatordate:iso8601)%3cad%3e%(authordate:iso8601)%3cs%3e%(subject)' usage: git tag [-a|-s|-u <key-id>] [-f] [-m <msg>|-F <file>] <tagname> [<head>] or: git tag -d <tagname>... or: git tag -l [-n[<num>]] [--contains <commit>] [--points-at <object>] [<pattern>...] or: git tag -v <tagname>... -l, --list list tag names -n[<n>] print <n> lines of each tag message -d, --delete delete tags -v, --verify verify tags Tag creation options -a, --annotate annotated tag, needs a message -m, --message <message> tag message -F, --file <file> read message from file ...

**Comment by eamodio** [comment]
GitLens needs Git version `2.7.2` or later to function properly. You should have seen a warning notification about that -- did you?

Also you said you upgraded recently -- what version were you using?

**Reply by mdear** [reply]
Thanks for the reply.  I see the problem.  GitLens was previously working in my WSL in which I have git 2.25.1.

When I tried to install it on my Linux server and run it via ssh Remote-SSH, it didn't work due to the old version of git that comes with the OS I was using.

I'll just use GitLens on my WSL going forward, I think.

**Reply by mdear** [reply]
And I don't recall seeing a warning notification when I installed GitLens on my Linux server, I just saw the notice that views had moved to the Source Control tab.
